src/Debug/skeletonViewer.ts

```diff
@@ -30,6 +33,253 @@ export class SkeletonViewer {
     /** public Display constants BABYLON.SkeletonViewer.DISPLAY_SPHERE_AND_SPURS */
     public static readonly DISPLAY_SPHERE_AND_SPURS = 2;
 
+    /** public static method to create a BoneWeight Shader
+     * @param options The constructor options
+     * @param scene The scene that the shader is scoped to
+     * @returns The created ShaderMaterial
+     */
+    static CreateBoneWeightShader(options: IBoneWeightShaderOptions, scene: Scene): ShaderMaterial {
+
+        let skeleton: Skeleton = options.skeleton;
+        let colorBase: Color3 = options.colorBase ?? Color3.Black();

```

Comment:
i would bind the color inside the skeletonviewer so we can change them anytime without rebuilding a shader

Reply:
for these ones you just do weightShader.setColor3('colorBase', color) and the shader will update.

Also for the skeletonMap one which uses the CreateColorBuffer thing as long as you do like
skeletonMap.setFloats('colorMap', SkeletonViewer.CreateColorBuffer(...)) it will update that as well without a rebuild.
That way the user has full control  or do you want it to be a global thing for all viewers like you are suggesting?

Ill have to make it clear in the docs either way.

Reply:
Well you are totally right. Then we can make the 3 statics private. And expose colors on the skeleton viewers as properties.
We just need to set them before rendering as we will share the shader between skeletonviewer instances

Reply:
Ok want me to leave this open or close it till I can do that?

Reply:
I'm wondering if that code:
```
shader.setColor3('colorBase', colorBase);
        shader.setColor3('colorZero', colorZero);
        shader.setColor3('colorQuarter', colorQuarter);
        shader.setColor3('colorHalf', colorHalf);
        shader.setColor3('colorFull', colorFull);
        shader.setFloat('targetBoneIndex', targetBoneIndex);
```
should not be done outside by the skeletonviewer
 So all viewer can share the same shader? or do you plan to get a shader per viewer?

Reply:
Anyhow I think this should be on the viewer. The static should compile and create the shadermaterial and that's it
interacting with the material is on the viewer

Reply:
(You can keep it open no problem)

Reply:
Additional note: please be sure to dispose everything when disposing the viewer

Reply:
> I'm wondering if that code:
> 
> ```
> shader.setColor3('colorBase', colorBase);
>         shader.setColor3('colorZero', colorZero);
>         shader.setColor3('colorQuarter', colorQuarter);
>         shader.setColor3('colorHalf', colorHalf);
>         shader.setColor3('colorFull', colorFull);
>         shader.setFloat('targetBoneIndex', targetBoneIndex);
> ```
> 
> should not be done outside by the skeletonviewer
> So all viewer can share the same shader? or do you plan to get a shader per viewer?

The shader is not necessarily just for the Viewer, its for any mesh with a skeleton.  I just nested it under the skeletonViewer as statics because its kinda like they are in a way another "View" of the skeleton.  The Viewer itself would up to the user to do like viewer.debugMesh.material = SkeletonViewer.CreateShader#(...).  Normally I dont think the viewer itself would have the shader but the mesh that the viewer is on would more often.

So I envisioned this kind usage from a user standpoint:
Import RiggedMesh
Create SkeletonViewer
Add a DebugShader to the RiggedMesh.

figured we would add and argument in the SkeletonViewer Options to automatically do this as well.

But for all general purposes the debugShaders can be used by any mesh.

Reply:
Well but it is hosted by the viewer class. if this is the case, it should live by itself so I do not need to instanciate or import the skeletonViewer to get the shader

But in this case We should probably create it as a material (like the normalMaterial):
https://github.com/BabylonJS/Babylon.js/tree/master/materialsLibrary/src/normal

DEfinitely better but a bit more work

Reply:
Sounds good, I will make those changes when I get back to town on Tuesday of next week!

Reply:
Enjoy buddy!

Reply:
I am back in town and will be getting back on this hopefully by the end of the day.  Gotta get caught up around the board here.